aten/src/ATen/native/vulkan/gen_spv.py

```diff
@@ -0,0 +1,142 @@
+    dirPath, fileName = filePath.rsplit('/', 1)
+    return fileName.replace("/", "_").replace(".", "_")
+
+def genCppH(hFilePath, cppFilePath, srcDirPath, glslcPath, tmpDirPath):
+    print("hFilePath:{} cppFilePath:{} srcDirPath:{} glslcPath:{} tmpDirPath:{}".format(
+        hFilePath, cppFilePath, srcDirPath, glslcPath, tmpDirPath))
+
+    cmd = "find " + srcDirPath + " -name \"vulkan*.glsl\""
+    vexs = os.popen(cmd).read().split('\n')
+    srcPaths = []
+    for f in vexs:
+      if len(f) > 1:
+        srcPaths.append(f)
+        srcPaths.sort()
+    print("srcPaths:{}".format(srcPaths))
+
+    spvPaths = []
+    for srcPath in srcPaths:
+        print("srcPath {}".format(srcPath))
+        name = getName(srcPath).replace("_glsl", "")
+        print("name {}".format(name))
+
+        spvPath = tmpDirPath + "/" + name + ".spv";
+        print("spvPath {}".format(spvPath))
+
+        cmd = "{} -fshader-stage=compute {} -o {} --target-env=vulkan1.0".format(
+            glslcPath, srcPath, spvPath)
+        print("\nglslc cmd: {}".format(cmd))
+
+        glslcProcess = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+        glslcProcess.wait()
+        print("returncode:{}\n".format(glslcProcess.returncode))
+        if glslcProcess.returncode != 0:
+          raise Exception("Error compiling glsl " + srcPath)
+        spvPaths.append(spvPath)
+
+    print("hFilePath:{}".format(hFilePath))
+    print("cppFilePath:{}".format(cppFilePath))
+    h = "#pragma once\n"
+    nsbegin = "\nnamespace at { namespace native { namespace vulkan { \n"
+    nsend = "\n} } } //namespace at::native::vulkan\n"
+
+    h += nsbegin
+
+    cpp = "#include <ATen/native/vulkan/{}>".format(H_NAME)
+    cpp += nsbegin
+
+    for spvPath in spvPaths:
+        name = getName(spvPath)
+        name_len = name + "_len"
+        h += "extern const unsigned char {}[];\n".format(name)
+        h += "extern unsigned int {};\n".format(name_len)
+
+        cpp += "const unsigned char " + name + "[] = {\n"
+        sizeBytes = 0
+        with open(spvPath, 'rb') as f:
+            line = ""
+            n = 0
+            while True:
+                byte = f.read(1)
+                if not byte:
+                    break
+                int_value = ord(byte)
+                s = "0x{0:02X},".format(int_value)
+                if n==16:
+                    cpp += line + "\n"
+                    line = ""
+                    n = 0
+                line += s
+                n += 1
+                sizeBytes += 1
+
+            if n > 0:
+                cpp += line + "\n"
+                line = ""
+                n = 0
+            cpp += "};\n"
+        cpp += "unsigned int {} = {};\n".format(name_len, sizeBytes)
+
+    cpp += nsend
+    h += nsend
+
+    with open(hFilePath, "w") as f:
+      f.write(h)
+    with open(cppFilePath, "w") as f:
+      f.write(cpp)
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser(description='')
```

Comment:
One consequence of putting code directly here is that variables like "parser" become visible from *every* function in this file, which is probably not what you want.  I prefer

```
def main(argv):
    # blah

if __name__ == "__main__":
    sys.exit(main(sys.argv))
```

Then all of my variables are local to main.